i18n: package securedrop-app translation files

dachary opened the pull request:
## Status

Ready for review

## Description of Changes

Add the i18n files to the app package so they are installed and useable.

## Testing

* make build-debs

Note that the build process will use ./manage.py --verbose translate to build the .mo file. This must be verified to run successfully. There are no translations yet so it will not produce any file. But it should not fail either.

## Deployment

Although the app package now contains more files, they are not used and are not concern for existing deployments.

## Checklist

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

dachary commented:
rebased & repushed

dachary commented:
@redshiftzero thanks for the carefully review. I'll prepare the next item :-)

dachary commented:
rebased & repushed to trim the **18n: manage: make translate --compile more pythonic** commit that was merged via another pull request.

redshiftzero commented:
Arg sorry @dachary - rebase?

dachary commented:
rebased & repushed :-)

conorsch commented on securedrop/translations/messages.pot:
```diff
@@ -0,0 +1,18 @@
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: SecureDrop 0.3.12\n"
```
I spy a hardcoded version string! If we cannot make this a Jinja var, then it must be added to the `./update_version.sh` script so that it gets incremented during the normal release process.

dachary replied:
This is a generated file (from ./manage.py translate) so it is updated every time it is re-generated.

The only reason for committing it is to communicate with the weblate server. It goes like this:

* strings changes in code and templates
* they are [committed to messages.pot](https://docs.securedrop.org/en/latest/development/i18n.html#updating-strings-to-be-translated)
* weblate.securedrop.club pulls the repository and updates its indices with the content of messages.pot

Does this seem like an acceptable workflow to you ?

conorsch commented on install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml:
```diff
@@ -0,0 +1,18 @@
+- name: Compile PO to MO.
+  shell: "cp config.py.example config.py ; trap 'rm config.py' EXIT ; PYTHONDONTWRITEBYTECODE=true ./manage.py --verbose translate --compile"
+  args:
+    chdir: "{{ securedrop_code_filtered }}"
```
Crafty! Hard to read, though. Seems like the desired control flow is:

1. Ensure that `config.py` exists.
2. Run the `manage.py translate --compile` command.
3. Remove the `config.py` file.

We could definitely break this up into separate tasks to help readability, but perhaps the functionality would be clear to future maintainers. Still, a reformatting of the current logic like so:

```
diff --git a/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml b/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml
index 0c9a574d..a4653733 100644
--- a/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml
+++ b/install_files/ansible-base/roles/build-securedrop-app-code-deb-pkg/tasks/translations.yml
@@ -13,6 +13,11 @@
     - pip
 
 - name: Compile PO to MO.
-  shell: "cp config.py.example config.py ; trap 'rm config.py' EXIT ; PYTHONDONTWRITEBYTECODE=true ./manage.py --verbose translate --compile"
+  shell: >-
+    cp config.py.example config.py ;
+    trap 'rm config.py' EXIT ;
+    ./manage.py --verbose translate --compile
   args:
     chdir: "{{ securedrop_code_filtered }}"
+  env:
+    PYTHONDONTWRITEBYTECODE: true
```

Strikes me as an improvement. (Note I haven't actually tested the patch above.)

dachary replied:
Much better to read indeed, thanks for the diff :-)

dachary commented:
> Will prioritize doing so next week.

Since code freeze is tuesday and monday is labor day, it is reasonable to say this commit missed the merge window. It would have been good to merge it, of course. But it's not a big deal if it is not.

dachary commented:
repushed with the suggested change

dachary commented:
> I do see a failing test:

Hum... I don't see how this failure can be related to this diff. Rebased & repushed. And updated the build instructions, good point !

dachary commented:
this tests pass now

conorsch commented:
:ok_hand: Looking good! Thanks for all the back-and-forth on this one, @dachary—glad it's finally going in!

dachary commented:
@conorsch would you like me to do something else before this can be merged ?
